teamcity: failed tests on master: test (proposer evaluated kv)/TestBackupRestoreBank

The following tests appear to have failed:

[#151144](https://teamcity.cockroachdb.com/viewLog.html?buildId=151144):

```
--- FAIL: test (proposer evaluated kv)/TestBackupRestoreBank (46.110s)
soon.go:43: ccl/sqlccl/backup_test.go:473, condition failed to evaluate within 45s: The bank is not in good order. Total failures: 1
------- Stdout: -------
W170215 16:02:24.125755 9046 server/status/runtime.go:116  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I170215 16:02:24.126790 9046 server/config.go:482  1 storage engine initialized
I170215 16:02:24.127031 9046 server/node.go:443  [n?] store [n0,s0] not bootstrapped
I170215 16:02:24.128984 9046 server/node.go:372  [n?] **** cluster b5f9a630-0816-47d2-a3eb-f2916afa02ac has been created
I170215 16:02:24.129003 9046 server/node.go:373  [n?] **** add additional nodes by specifying --join=127.0.0.1:45838
I170215 16:02:24.129521 9046 storage/store.go:1261  [n1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I170215 16:02:24.129545 9046 server/node.go:456  [n1] initialized store [n1,s1]: {Capacity:536870912 Available:536870912 RangeCount:1 LeaseCount:0}
I170215 16:02:24.129567 9046 server/node.go:341  [n1] node ID 1 initialized
I170215 16:02:24.129605 9046 gossip/gossip.go:293  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:45838" > attrs:<> locality:<> 
I170215 16:02:24.129751 9046 storage/stores.go:296  [n1] read 0 node addresses from persistent storage
I170215 16:02:24.129785 9046 server/node.go:605  [n1] connecting to gossip network to verify cluster ID...
I170215 16:02:24.130692 9046 server/node.go:629  [n1] node connected via gossip and verified as part of cluster "b5f9a630-0816-47d2-a3eb-f2916afa02ac"
I170215 16:02:24.130716 9046 server/node.go:391  [n1] node=1: started with [[]=] engine(s) and attributes []
I170215 16:02:24.130739 9046 sql/executor.go:333  [n1] creating distSQLPlanner with address {tcp 127.0.0.1:45838}
I170215 16:02:24.131840 9135 storage/split_queue.go:98  [split,n1,s1,r1/1:/M{in-ax},@c42182f180] splitting at key /Table/0/0
I170215 16:02:24.132649 9135 storage/replica_command.go:2397  [split,n1,s1,r1/1:/M{in-ax},@c42182f180] initiating a split of this range at key /Table/0 [r2]
I170215 16:02:24.132765 9046 server/server.go:644  [n1] starting https server at 127.0.0.1:46688
I170215 16:02:24.132781 9046 server/server.go:645  [n1] starting grpc/postgres server at 127.0.0.1:45838
I170215 16:02:24.132792 9046 server/server.go:646  [n1] advertising CockroachDB node at 127.0.0.1:45838
I170215 16:02:24.141186 9135 storage/split_queue.go:98  [split,n1,s1,r2/1:/{Table/0-Max},@c423d06a80] splitting at key /Table/11/0
I170215 16:02:24.141227 9135 storage/replica_command.go:2397  [split,n1,s1,r2/1:/{Table/0-Max},@c423d06a80] initiating a split of this range at key /Table/11 [r3]
I170215 16:02:24.141597 9145 sql/event_log.go:95  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:45838} Attrs: Locality:} ClusterID:b5f9a630-0816-47d2-a3eb-f2916afa02ac StartedAt:1487174544130704116 LastUp:1487174544130704116}
I170215 16:02:24.145553 9135 storage/split_queue.go:98  [split,n1,s1,r3/1:/{Table/11-Max},@c4216a3c00] splitting at key /Table/12/0
I170215 16:02:24.145588 9135 storage/replica_command.go:2397  [split,n1,s1,r3/1:/{Table/11-Max},@c4216a3c00] initiating a split of this range at key /Table/12 [r4]
I170215 16:02:24.147839 9046 sql/event_log.go:95  [n1] Event: "alter_table", target: 12, info: {TableName:eventlog Statement:ALTER TABLE system.eventlog ALTER COLUMN uniqueID SET DEFAULT uuid_v4() User:node MutationID:0 CascadeDroppedViews:[]}
I170215 16:02:24.152626 9135 storage/split_queue.go:98  [split,n1,s1,r4/1:/{Table/12-Max},@c42081aa80] splitting at key /Table/13/0
I170215 16:02:24.152663 9135 storage/replica_command.go:2397  [split,n1,s1,r4/1:/{Table/12-Max},@c42081aa80] initiating a split of this range at key /Table/13 [r5]
I170215 16:02:24.156601 9135 storage/split_queue.go:98  [split,n1,s1,r5/1:/{Table/13-Max},@c4204cca80] splitting at key /Table/14/0
I170215 16:02:24.156642 9135 storage/replica_command.go:2397  [split,n1,s1,r5/1:/{Table/13-Max},@c4204cca80] initiating a split of this range at key /Table/14 [r6]
I170215 16:02:24.161581 9046 server/server.go:771  [n1] done ensuring all necessary migrations have run
I170215 16:02:24.161604 9046 server/server.go:773  [n1] serving sql connections
W170215 16:02:24.164439 9046 gossip/gossip.go:1178  [n?] no incoming or outgoing connections
W170215 16:02:24.169854 9046 server/status/runtime.go:116  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I170215 16:02:24.171059 9046 server/config.go:482  1 storage engine initialized
I170215 16:02:24.171346 9046 server/node.go:443  [n?] store [n0,s0] not bootstrapped
I170215 16:02:24.171367 9046 storage/stores.go:296  [n?] read 0 node addresses from persistent storage
I170215 16:02:24.171389 9046 server/node.go:605  [n?] connecting to gossip network to verify cluster ID...
I170215 16:02:24.180734 9319 gossip/client.go:131  [n?] started gossip client to 127.0.0.1:45838
I170215 16:02:24.180980 9339 gossip/server.go:214  [n1] received initial cluster-verification connection from {tcp 127.0.0.1:38650}
I170215 16:02:24.181286 9340 storage/stores.go:312  [n?] wrote 1 node addresses to persistent storage
I170215 16:02:24.181372 9046 server/node.go:629  [n?] node connected via gossip and verified as part of cluster "b5f9a630-0816-47d2-a3eb-f2916afa02ac"
I170215 16:02:24.189968 9046 kv/dist_sender.go:367  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I170215 16:02:24.191614 9046 server/node.go:334  [n?] new node allocated ID 2
I170215 16:02:24.191666 9046 gossip/gossip.go:293  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:38650" > attrs:<> locality:<> 
I170215 16:02:24.191730 9046 server/node.go:391  [n2] node=2: started with [[]=] engine(s) and attributes []
I170215 16:02:24.191758 9046 sql/executor.go:333  [n2] creating distSQLPlanner with address {tcp 127.0.0.1:38650}
I170215 16:02:24.193605 9046 server/server.go:644  [n2] starting https server at 127.0.0.1:43228
I170215 16:02:24.193622 9046 server/server.go:645  [n2] starting grpc/postgres server at 127.0.0.1:38650
I170215 16:02:24.193629 9046 server/server.go:646  [n2] advertising CockroachDB node at 127.0.0.1:38650
I170215 16:02:24.198857 9238 storage/stores.go:312  [n1] wrote 1 node addresses to persistent storage
I170215 16:02:24.199126 9046 server/server.go:771  [n2] done ensuring all necessary migrations have run
I170215 16:02:24.199147 9046 server/server.go:773  [n2] serving sql connections
I170215 16:02:24.200066 9378 server/node.go:586  [n2] bootstrapped store [n2,s2]
I170215 16:02:24.201069 9036 storage/replica_raftstorage.go:414  [replicate,n1,s1,r1/1:/{Min-Table/0},@c42182f180] generated preemptive snapshot e42d9a57 at index 42
W170215 16:02:24.204392 9046 gossip/gossip.go:1178  [n?] no incoming or outgoing connections
I170215 16:02:24.204897 9381 sql/event_log.go:95  [n2] Event: "node_join", target: 2, info: {Descriptor:{NodeID:2 Address:{NetworkField:tcp AddressField:127.0.0.1:38650} Attrs: Locality:} ClusterID:b5f9a630-0816-47d2-a3eb-f2916afa02ac StartedAt:1487174544191714799 LastUp:1487174544191714799}
W170215 16:02:24.206351 9046 server/status/runtime.go:116  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I170215 16:02:24.208361 9046 server/config.go:482  1 storage engine initialized
I170215 16:02:24.208707 9046 server/node.go:443  [n?] store [n0,s0] not bootstrapped
I170215 16:02:24.208727 9046 storage/stores.go:296  [n?] read 0 node addresses from persistent storage
I170215 16:02:24.208751 9046 server/node.go:605  [n?] connecting to gossip network to verify cluster ID...
I170215 16:02:24.225963 9408 gossip/client.go:131  [n?] started gossip client to 127.0.0.1:45838
I170215 16:02:24.226454 9422 gossip/server.go:214  [n1] received initial cluster-verification connection from {tcp 127.0.0.1:36086}
I170215 16:02:24.226614 9046 server/node.go:629  [n?] node connected via gossip and verified as part of cluster "b5f9a630-0816-47d2-a3eb-f2916afa02ac"
I170215 16:02:24.226900 9516 storage/stores.go:312  [n?] wrote 1 node addresses to persistent storage
I170215 16:02:24.226934 9516 storage/stores.go:312  [n?] wrote 2 node addresses to persistent storage
I170215 16:02:24.226963 9036 storage/store.go:3312  [replicate,n1,s1,r1/1:/{Min-Table/0},@c42182f180] streamed snapshot: kv pairs: 574, log entries: 32, 8ms
I170215 16:02:24.227230 9495 storage/replica_raftstorage.go:596  [n2,s2,r1/?:{-},@c423be1c00] applying preemptive snapshot at index 42 (id=e42d9a57, encoded size=253175, 1 rocksdb batches, 32 log entries)
I170215 16:02:24.238153 9495 storage/replica_raftstorage.go:604  [n2,s2,r1/?:/{Min-Table/0},@c423be1c00] applied preemptive snapshot in 11ms [clear=0ms batch=0ms entries=10ms commit=1ms]
I170215 16:02:24.240238 9036 storage/replica_command.go:3253  [replicate,n1,s1,r1/1:/{Min-Table/0},@c42182f180] change replicas (remove {2 2 2}): read existing descriptor range_id:1 start_key:"" end_key:"\210" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
I170215 16:02:24.240528 9046 kv/dist_sender.go:367  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I170215 16:02:24.241083 9046 server/node.go:334  [n?] new node allocated ID 3
I170215 16:02:24.241139 9046 gossip/gossip.go:293  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:36086" > attrs:<> locality:<> 
I170215 16:02:24.241188 9046 server/node.go:391  [n3] node=3: started with [[]=] engine(s) and attributes []
I170215 16:02:24.241213 9046 sql/executor.go:333  [n3] creating distSQLPlanner with address {tcp 127.0.0.1:36086}
I170215 16:02:24.243636 9046 server/server.go:644  [n3] starting https server at 127.0.0.1:33083
I170215 16:02:24.243652 9046 server/server.go:645  [n3] starting grpc/postgres server at 127.0.0.1:36086
I170215 16:02:24.243664 9046 server/server.go:646  [n3] advertising CockroachDB node at 127.0.0.1:36086
I170215 16:02:24.244416 9532 storage/replica.go:2487  [n1,s1,r1/1:/{Min-Table/0},@c42182f180] proposing ADD_REPLICA {NodeID:2 StoreID:2 ReplicaID:2}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2}]
I170215 16:02:24.244585 9471 storage/stores.go:312  [n1] wrote 2 node addresses to persistent storage
I170215 16:02:24.244797 9472 storage/stores.go:312  [n2] wrote 2 node addresses to persistent storage
I170215 16:02:24.245118 9036 storage/replica_raftstorage.go:414  [replicate,n1,s1,r2/1:/Table/{0-11},@c423d06a80] generated preemptive snapshot b57833be at index 25
I170215 16:02:24.245811 9036 storage/store.go:3312  [replicate,n1,s1,r2/1:/Table/{0-11},@c423d06a80] streamed snapshot: kv pairs: 33, log entries: 15, 1ms
I170215 16:02:24.246294 9605 storage/raft_transport.go:437  [n2] raft transport stream to node 1 established
I170215 16:02:24.246329 9537 storage/replica_raftstorage.go:596  [n2,s2,r2/?:{-},@c42317d180] applying preemptive snapshot at index 25 (id=b57833be, encoded size=23765, 1 rocksdb batches, 15 log entries)
I170215 16:02:24.246639 9537 storage/replica_raftstorage.go:604  [n2,s2,r2/?:/Table/{0-11},@c42317d180] applied preemptive snapshot in 0ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I170215 16:02:24.247968 9036 storage/replica_command.go:3253  [replicate,n1,s1,r2/1:/Table/{0-11},@c423d06a80] change replicas (remove {2 2 2}): read existing descriptor range_id:2 start_key:"\210" end_key:"\223" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
I170215 16:02:24.248811 9497 server/node.go:586  [n3] bootstrapped store [n3,s3]
I170215 16:02:24.249801 9500 sql/event_log.go:95  [n3] Event: "node_join", target: 3, info: {Descriptor:{NodeID:3 Address:{NetworkField:tcp AddressField:127.0.0.1:36086} Attrs: Locality:} ClusterID:b5f9a630-0816-47d2-a3eb-f2916afa02ac StartedAt:1487174544241173983 LastUp:1487174544241173983}
I170215 16:02:24.250318 9046 server/server.go:771  [n3] done ensuring all necessary migrations have run
I170215 16:02:24.250339 9046 server/server.go:773  [n3] serving sql connections
I170215 16:02:24.252238 9642 storage/replica.go:2487  [n1,s1,r2/1:/Table/{0-11},@c423d06a80] proposing ADD_REPLICA {NodeID:2 StoreID:2 ReplicaID:2}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2}]
I170215 16:02:24.252752 9036 storage/replica_raftstorage.go:414  [replicate,n1,s1,r3/1:/Table/1{1-2},@c4216a3c00] generated preemptive snapshot 86799fb0 at index 18
I170215 16:02:24.267759 9036 storage/store.go:3312  [replicate,n1,s1,r3/1:/Table/1{1-2},@c4216a3c00] streamed snapshot: kv pairs: 10, log entries: 8, 1ms
I170215 16:02:24.268276 9689 storage/replica_raftstorage.go:596  [n3,s3,r3/?:{-},@c4221b1180] applying preemptive snapshot at index 18 (id=86799fb0, encoded size=10510, 1 rocksdb batches, 8 log entries)
I170215 16:02:24.268547 9689 storage/replica_raftstorage.go:604  [n3,s3,r3/?:/Table/1{1-2},@c4221b1180] applied preemptive snapshot in 0ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I170215 16:02:24.268990 9036 storage/replica_command.go:3253  [replicate,n1,s1,r3/1:/Table/1{1-2},@c4216a3c00] change replicas (remove {3 3 2}): read existing descriptor range_id:3 start_key:"\223" end_key:"\224" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
I170215 16:02:24.273545 9553 storage/replica.go:2487  [n1,s1,r3/1:/Table/1{1-2},@c4216a3c00] proposing ADD_REPLICA {NodeID:3 StoreID:3 ReplicaID:2}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:3 StoreID:3 ReplicaID:2}]
I170215 16:02:24.274482 9036 storage/replica_raftstorage.go:414  [replicate,n1,s1,r6/1:/{Table/14-Max},@c42088d180] generated preemptive snapshot 7aac1e00 at index 11
I170215 16:02:24.276835 9036 storage/store.go:3312  [replicate,n1,s1,r6/1:/{Table/14-Max},@c42088d180] streamed snapshot: kv pairs: 9, log entries: 1, 2ms
I170215 16:02:24.278413 9692 storage/raft_transport.go:437  [n3] raft transport stream to node 1 established
I170215 16:02:24.278673 9736 storage/replica_raftstorage.go:596  [n3,s3,r6/?:{-},@c422e07180] applying preemptive snapshot at index 11 (id=7aac1e00, encoded size=504, 1 rocksdb batches, 1 log entries)
I170215 16:02:24.278843 9736 storage/replica_raftstorage.go:604  [n3,s3,r6/?:/{Table/14-Max},@c422e07180] applied preemptive snapshot in 0ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I170215 16:02:24.279632 9036 storage/replica_command.go:3253  [replicate,n1,s1,r6/1:/{Table/14-Max},@c42088d180] change replicas (remove {3 3 2}): read existing descriptor range_id:6 start_key:"\226" end_key:"\377\377" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
I170215 16:02:24.284230 9719 storage/replica.go:2487  [n1,s1,r6/1:/{Table/14-Max},@c42088d180] proposing ADD_REPLICA {NodeID:3 StoreID:3 ReplicaID:2}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:3 StoreID:3 ReplicaID:2}]
I170215 16:02:24.284751 9036 storage/replica_raftstorage.go:414  [replicate,n1,s1,r4/1:/Table/1{2-3},@c42081aa80] generated preemptive snapshot 6f38aeb0 at index 20
I170215 16:02:24.286560 9726 storage/replica_raftstorage.go:596  [n2,s2,r4/?:{-},@c4238f5880] applying preemptive snapshot at index 20 (id=6f38aeb0, encoded size=16924, 1 rocksdb batches, 10 log entries)
I170215 16:02:24.286820 9726 storage/replica_raftstorage.go:604  [n2,s2,r4/?:/Table/1{2-3},@c4238f5880] applied preemptive snapshot in 0ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I170215 16:02:24.287279 9036 storage/store.go:3312  [replicate,n1,s1,r4/1:/Table/1{2-3},@c42081aa80] streamed snapshot: kv pairs: 30, log entries: 10, 2ms
I170215 16:02:24.287716 9036 storage/replica_command.go:3253  [replicate,n1,s1,r4/1:/Table/1{2-3},@c42081aa80] change replicas (remove {2 2 2}): read existing descriptor range_id:4 start_key:"\224" end_key:"\225" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
I170215 16:02:24.291689 9727 storage/replica.go:2487  [n1,s1,r4/1:/Table/1{2-3},@c42081aa80] proposing ADD_REPLICA {NodeID:2 StoreID:2 ReplicaID:2}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2}]
I170215 16:02:24.292218 9036 storage/replica_raftstorage.go:414  [replicate,n1,s1,r5/1:/Table/1{3-4},@c4204cca80] generated preemptive snapshot e48a0393 at index 25
I170215 16:02:24.293742 9036 storage/store.go:3312  [replicate,n1,s1,r5/1:/Table/1{3-4},@c4204cca80] streamed snapshot: kv pairs: 70, log entries: 15, 1ms
I170215 16:02:24.294016 9767 storage/replica_raftstorage.go:596  [n3,s3,r5/?:{-},@c42387e000] applying preemptive snapshot at index 25 (id=e48a0393, encoded size=27436, 1 rocksdb batches, 15 log entries)
I170215 16:02:24.294363 9767 storage/replica_raftstorage.go:604  [n3,s3,r5/?:/Table/1{3-4},@c42387e000] applied preemptive snapshot in 0ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I170215 16:02:24.294777 9036 storage/replica_command.go:3253  [replicate,n1,s1,r5/1:/Table/1{3-4},@c4204cca80] change replicas (remove {3 3 2}): read existing descriptor range_id:5 start_key:"\225" end_key:"\226" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
I170215 16:02:24.298196 9451 storage/replica.go:2487  [n1,s1,r5/1:/Table/1{3-4},@c4204cca80] proposing ADD_REPLICA {NodeID:3 StoreID:3 ReplicaID:2}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:3 StoreID:3 ReplicaID:2}]
I170215 16:02:24.298670 9036 storage/queue.go:693  [n1,replicate] purgatory is now empty
I170215 16:02:24.298987 9136 storage/replica_raftstorage.go:414  [replicate,n1,s1,r1/1:/{Min-Table/0},@c42182f180] generated preemptive snapshot 71d1e339 at index 61
I170215 16:02:24.308069 9136 storage/store.go:3312  [replicate,n1,s1,r1/1:/{Min-Table/0},@c42182f180] streamed snapshot: kv pairs: 787, log entries: 51, 9ms
I170215 16:02:24.308418 9455 storage/replica_raftstorage.go:596  [n3,s3,r1/?:{-},@c421e1ce00] applying preemptive snapshot at index 61 (id=71d1e339, encoded size=409978, 1 rocksdb batches, 51 log entries)
I170215 16:02:24.310277 9455 storage/replica_raftstorage.go:604  [n3,s3,r1/?:/{Min-Table/0},@c421e1ce00] applied preemptive snapshot in 2ms [clear=0ms batch=0ms entries=1ms commit=1ms]
I170215 16:02:24.311074 9136 storage/replica_command.go:3253  [replicate,n1,s1,r1/1:/{Min-Table/0},@c42182f180] change replicas (remove {3 3 3}): read existing descriptor range_id:1 start_key:"" end_key:"\210" replicas:<node_id:1 store_id:1 replica_id:1 > replicas:<node_id:2 store_id:2 replica_id:2 > next_replica_id:3 
I170215 16:02:24.322313 9826 storage/replica.go:2487  [n1,s1,r1/1:/{Min-Table/0},@c42182f180] proposing ADD_REPLICA {NodeID:3 StoreID:3 ReplicaID:3}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2} {NodeID:3 StoreID:3 ReplicaID:3}]
I170215 16:02:24.323650 9136 storage/replica_raftstorage.go:414  [replicate,n1,s1,r2/1:/Table/{0-11},@c423d06a80] generated preemptive snapshot ed0e714d at index 28
I170215 16:02:24.324631 9136 storage/store.go:3312  [replicate,n1,s1,r2/1:/Table/{0-11},@c423d06a80] streamed snapshot: kv pairs: 34, log entries: 18, 1ms
I170215 16:02:24.324966 9802 storage/replica_raftstorage.go:596  [n3,s3,r2/?:{-},@c423db7c00] applying preemptive snapshot at index 28 (id=ed0e714d, encoded size=26733, 1 rocksdb batches, 18 log entries)
I170215 16:02:24.325249 9802 storage/replica_raftstorage.go:604  [n3,s3,r2/?:/Table/{0-11},@c423db7c00] applied preemptive snapshot in 0ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I170215 16:02:24.325638 9136 storage/replica_command.go:3253  [replicate,n1,s1,r2/1:/Table/{0-11},@c423d06a80] change replicas (remove {3 3 3}): read existing descriptor range_id:2 start_key:"\210" end_key:"\223" replicas:<node_id:1 store_id:1 replica_id:1 > replicas:<node_id:2 store_id:2 replica_id:2 > next_replica_id:3 
I170215 16:02:24.329062 9787 storage/replica.go:2487  [n1,s1,r2/1:/Table/{0-11},@c423d06a80] proposing ADD_REPLICA {NodeID:3 StoreID:3 ReplicaID:3}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2} {NodeID:3 StoreID:3 ReplicaID:3}]
I170215 16:02:24.329976 9136 storage/replica_raftstorage.go:414  [replicate,n1,s1,r5/1:/Table/1{3-4},@c4204cca80] generated preemptive snapshot eeacea3a at index 33
I170215 16:02:24.336615 9136 storage/store.go:3312  [replicate,n1,s1,r5/1:/Table/1{3-4},@c4204cca80] streamed snapshot: kv pairs: 86, log entries: 23, 7ms
I170215 16:02:24.336920 9585 storage/replica_raftstorage.go:596  [n2,s2,r5/?:{-},@c423a15500] applying preemptive snapshot at index 33 (id=eeacea3a, encoded size=40349, 1 rocksdb batches, 23 log entries)
I170215 16:02:24.337426 9585 storage/replica_raftstorage.go:604  [n2,s2,r5/?:/Table/1{3-4},@c423a15500] applied preemptive snapshot in 0ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I170215 16:02:24.337832 9136 storage/replica_command.go:3253  [replicate,n1,s1,r5/1:/Table/1{3-4},@c4204cca80] change replicas (remove {2 2 3}): read existing descriptor range_id:5 start_key:"\225" end_key:"\226" replicas:<node_id:1 store_id:1 replica_id:1 > replicas:<node_id:3 store_id:3 replica_id:2 > next_replica_id:3 
I170215 16:02:24.341268 9824 storage/replica.go:2487  [n1,s1,r5/1:/Table/1{3-4},@c4204cca80] proposing ADD_REPLICA {NodeID:2 StoreID:2 ReplicaID:3}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:3 StoreID:3 ReplicaID:2} {NodeID:2 StoreID:2 ReplicaID:3}]
I170215 16:02:24.343317 9136 storage/replica_raftstorage.go:414  [replicate,n1,s1,r4/1:/Table/1{2-3},@c42081aa80] generated preemptive snapshot c5aeeccd at index 23
I170215 16:02:24.343969 9136 storage/store.go:3312  [replicate,n1,s1,r4/1:/Table/1{2-3},@c42081aa80] streamed snapshot: kv pairs: 31, log entries: 13, 1ms
I170215 16:02:24.344314 9862 storage/replica_raftstorage.go:596  [n3,s3,r4/?:{-},@c423d07180] applying preemptive snapshot at index 23 (id=c5aeeccd, encoded size=19892, 1 rocksdb batches, 13 log entries)
I170215 16:02:24.346229 9862 storage/replica_raftstorage.go:604  [n3,s3,r4/?:/Table/1{2-3},@c423d07180] applied preemptive snapshot in 2ms [clear=2ms batch=0ms entries=0ms commit=0ms]
I170215 16:02:24.346640 9136 storage/replica_command.go:3253  [replicate,n1,s1,r4/1:/Table/1{2-3},@c42081aa80] change replicas (remove {3 3 3}): read existing descriptor range_id:4 start_key:"\224" end_key:"\225" replicas:<node_id:1 store_id:1 replica_id:1 > replicas:<node_id:2 store_id:2 replica_id:2 > next_replica_id:3 
I170215 16:02:24.349939 9906 storage/replica.go:2487  [n1,s1,r4/1:/Table/1{2-3},@c42081aa80] proposing ADD_REPLICA {NodeID:3 StoreID:3 ReplicaID:3}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2} {NodeID:3 StoreID:3 ReplicaID:3}]
I170215 16:02:24.351913 9136 storage/replica_raftstorage.go:414  [replicate,n1,s1,r6/1:/{Table/14-Max},@c42088d180] generated preemptive snapshot 56f0eae9 at index 14
I170215 16:02:24.352650 9136 storage/store.go:3312  [replicate,n1,s1,r6/1:/{Table/14-Max},@c42088d180] streamed snapshot: kv pairs: 10, log entries: 4, 1ms
I170215 16:02:24.352921 9866 storage/replica_raftstorage.go:596  [n2,s2,r6/?:{-},@c423966380] applying preemptive snapshot at index 14 (id=56f0eae9, encoded size=3479, 1 rocksdb batches, 4 log entries)
I170215 16:02:24.353108 9866 storage/replica_raftstorage.go:604  [n2,s2,r6/?:/{Table/14-Max},@c423966380] applied preemptive snapshot in 0ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I170215 16:02:24.353849 9136 storage/replica_command.go:3253  [replicate,n1,s1,r6/1:/{Table/14-Max},@c42088d180] change replicas (remove {2 2 3}): read existing descriptor range_id:6 start_key:"\226" end_key:"\377\377" replicas:<node_id:1 store_id:1 replica_id:1 > replicas:<node_id:3 store_id:3 replica_id:2 > next_replica_id:3 
I170215 16:02:24.357016 9909 storage/replica.go:2487  [n1,s1,r6/1:/{Table/14-Max},@c42088d180] proposing ADD_REPLICA {NodeID:2 StoreID:2 ReplicaID:3}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:3 StoreID:3 ReplicaID:2} {NodeID:2 StoreID:2 ReplicaID:3}]
I170215 16:02:24.358954 9136 storage/replica_raftstorage.go:414  [replicate,n1,s1,r3/1:/Table/1{1-2},@c4216a3c00] generated preemptive snapshot 589b40f6 at index 21
I170215 16:02:24.359612 9136 storage/store.go:3312  [replicate,n1,s1,r3/1:/Table/1{1-2},@c4216a3c00] streamed snapshot: kv pairs: 11, log entries: 11, 1ms
I170215 16:02:24.359802 9632 storage/replica_raftstorage.go:596  [n2,s2,r3/?:{-},@c4239d4700] applying preemptive snapshot at index 21 (id=589b40f6, encoded size=13471, 1 rocksdb batches, 11 log entries)
I170215 16:02:24.360035 9632 storage/replica_raftstorage.go:604  [n2,s2,r3/?:/Table/1{1-2},@c4239d4700] applied preemptive snapshot in 0ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I170215 16:02:24.361132 9136 storage/replica_command.go:3253  [replicate,n1,s1,r3/1:/Table/1{1-2},@c4216a3c00] change replicas (remove {2 2 3}): read existing descriptor range_id:3 start_key:"\223" end_key:"\224" replicas:<node_id:1 store_id:1 replica_id:1 > replicas:<node_id:3 store_id:3 replica_id:2 > next_replica_id:3 
I170215 16:02:24.365496 9960 storage/replica.go:2487  [n1,s1,r3/1:/Table/1{1-2},@c4216a3c00] proposing ADD_REPLICA {NodeID:2 StoreID:2 ReplicaID:3}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:3 StoreID:3 ReplicaID:2} {NodeID:2 StoreID:2 ReplicaID:3}]
I170215 16:02:24.473657 9971 sql/event_log.go:95  [client=127.0.0.1:58788,user=root,n1] Event: "create_database", target: 50, info: {DatabaseName:bench Statement:CREATE DATABASE bench User:root}
I170215 16:02:24.473997 9135 storage/split_queue.go:98  [split,n1,s1,r6/1:/{Table/14-Max},@c42088d180] splitting at key /Table/50/0
I170215 16:02:24.474035 9135 storage/replica_command.go:2397  [split,n1,s1,r6/1:/{Table/14-Max},@c42088d180] initiating a split of this range at key /Table/50 [r7]
I170215 16:02:24.480828 9971 sql/event_log.go:95  [client=127.0.0.1:58788,user=root,n1] Event: "create_table", target: 51, info: {TableName:bench.bank Statement:CREATE TABLE bench.bank (id INT PRIMARY KEY, balance INT, payload STRING, FAMILY (id, balance, payload)) User:root}
I170215 16:02:24.485558 9991 storage/replica_command.go:2397  [n1,s1,r6/1:/{Table/14-Max},@c42088d180] initiating a split of this range at key /Table/51/1/1 [r8]
I170215 16:02:24.496781 10052 storage/replica_command.go:2397  [n1,s1,r8/1:/{Table/51/1/1-Max},@c4217e2a80] initiating a split of this range at key /Table/51/1/2 [r9]
I170215 16:02:24.506411 10100 storage/replica_command.go:2397  [n1,s1,r9/1:/{Table/51/1/2-Max},@c421c4c380] initiating a split of this range at key /Table/51/1/3 [r10]
I170215 16:02:24.514722 10124 storage/replica_command.go:2397  [n1,s1,r10/1:/{Table/51/1/3-Max},@c423572000] initiating a split of this range at key /Table/51/1/4 [r11]
I170215 16:02:24.523895 10092 storage/raft_transport.go:437  [n3] raft transport stream to node 2 established
I170215 16:02:24.524506 10139 storage/replica_command.go:2397  [n1,s1,r11/1:/{Table/51/1/4-Max},@c421965c00] initiating a split of this range at key /Table/51/1/5 [r12]
I170215 16:02:24.534295 10141 storage/replica_command.go:2397  [n1,s1,r12/1:/{Table/51/1/5-Max},@c420aa4a80] initiating a split of this range at key /Table/51/1/6 [r13]
I170215 16:02:24.536742 10168 storage/raft_transport.go:437  [n2] raft transport stream to node 3 established
I170215 16:02:24.542501 10279 storage/replica_command.go:2397  [n1,s1,r13/1:/{Table/51/1/6-Max},@c420aa5180] initiating a split of this range at key /Table/51/1/7 [r14]
E170215 16:02:24.542986 9135 storage/queue.go:628  [split,n1,s1,r6/1:/Table/{14-51/1/1},@c42088d180] unable to split [n1,s1,r6/1:/Table/{14-51/1/1}] at key "/Table/50/0": unexpected value: raw_bytes:"\330G\334<\003\010\006\022\001\226\032\003\273\211\211\"\006\010\001\020\001\030\001\"\006\010\003\020\003\030\002\"\006\010\002\020\002\030\003(\004" timestamp:<wall_time:1487174544485590926 logical:0 > 
I170215 16:02:24.543070 9135 storage/split_queue.go:98  [split,n1,s1,r6/1:/Table/{14-51/1/1},@c42088d180] splitting at key /Table/50/0
I170215 16:02:24.543103 9135 storage/replica_command.go:2397  [split,n1,s1,r6/1:/Table/{14-51/1/1},@c42088d180] initiating a split of this range at key /Table/50 [r15]
I170215 16:02:24.561860 9135 storage/split_queue.go:98  [split,n1,s1,r15/1:/Table/5{0-1/1/1},@c420266a80] splitting at key /Table/51/0
I170215 16:02:24.561909 9135 storage/replica_command.go:2397  [split,n1,s1,r15/1:/Table/5{0-1/1/1},@c420266a80] initiating a split of this range at key /Table/51 [r16]
I170215 16:02:24.566702 10349 storage/replica_command.go:2397  [n1,s1,r14/1:/{Table/51/1/7-Max},@c421a88e00] initiating a split of this range at key /Table/51/1/8 [r17]
I170215 16:02:24.576846 9139 storage/replica_raftstorage.go:414  [raftsnapshot,n1,s1,r14/1:/{Table/51/1/7-Max},@c421a88e00] generated Raft snapshot 2ccc93c8 at index 13
I170215 16:02:24.578986 9139 storage/store.go:3312  [raftsnapshot,n1,s1,r14/1:/{Table/51/1/7-Max},@c421a88e00] streamed snapshot: kv pairs: 15, log entries: 3, 2ms
I170215 16:02:24.579456 10375 storage/replica_raftstorage.go:596  [n2,s2,r14/3:/{Table/51/1/7-Max},@c423d06000] applying Raft snapshot at index 13 (id=2ccc93c8, encoded size=3416, 1 rocksdb batches, 3 log entries)
I170215 16:02:24.580127 10375 storage/replica_raftstorage.go:604  [n2,s2,r14/3:/{Table/51/1/7-Max},@c423d06000] applied Raft snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I170215 16:02:24.583639 9139 storage/replica_raftstorage.go:414  [raftsnapshot,n1,s1,r15/1:/Table/5{0-1},@c420266a80] generated Raft snapshot ed532803 at index 15
I170215 16:02:24.584483 10387 storage/replica_raftstorage.go:596  [n2,s2,r15/3:/Table/5{0-1/1/1},@c4211ba700] applying Raft snapshot at index 15 (id=ed532803, encoded size=5932, 1 rocksdb batches, 5 log entries)
I170215 16:02:24.584708 10387 storage/replica_raftstorage.go:604  [n2,s2,r15/3:/Table/5{0-1},@c4211ba700] applied Raft snapshot in 0ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I170215 16:02:24.585670 10404 storage/replica_command.go:2397  [n1,s1,r17/1:/{Table/51/1/8-Max},@c4211bb180] initiating a split of this range at key /Table/51/1/9 [r18]
I170215 16:02:24.586026 9139 storage/store.go:3312  [raftsnapshot,n1,s1,r15/1:/Table/5{0-1},@c420266a80] streamed snapshot: kv pairs: 11, log entries: 5, 1ms
I170215 16:02:24.619490 9139 storage/replica_raftstorage.go:414  [raftsnapshot,n1,s1,r16/1:/Table/51{-/1/1},@c42088dc00] generated Raft snapshot 97ada000 at index 10
I170215 16:02:24.619968 9139 storage/store.go:3312  [raftsnapshot,n1,s1,r16/1:/Table/51{-/1/1},@c42088dc00] streamed snapshot: kv pairs: 10, log entries: 0, 0ms
I170215 16:02:24.620122 10440 storage/replica_raftstorage.go:596  [n2,s2,r16/3:{-},@c422d30a80] applying Raft snapshot at index 10 (id=97ada000, encoded size=641, 1 rocksdb batches, 0 log entries)
I170215 16:02:24.620314 10440 storage/replica_raftstorage.go:604  [n2,s2,r16/3:/Table/51{-/1/1},@c422d30a80] applied Raft snapshot in 0ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I170215 16:02:24.760641 11500 storage/replica_command.go:2397  [n1,s1,r18/1:/{Table/51/1/9-Max},@c42317d880] initiating a split of this range at key /Table/52/1/5 [r19]
I170215 16:02:24.767836 11528 storage/replica_command.go:2397  [n1,s1,r19/1:/{Table/52/1/5-Max},@c421f18e00] initiating a split of this range at key /Table/52/1/8 [r20]
I170215 16:02:24.767865 11558 storage/replica_command.go:2397  [n1,s1,r18/1:/Table/5{1/1/9-2/1/5},@c42317d880] initiating a split of this range at key /Table/52/1/2 [r21]
I170215 16:02:24.775311 11518 storage/replica_command.go:2397  [n1,s1,r21/1:/Table/52/1/{2-5},@c421e0d180] initiating a split of this range at key /Table/52/1/4 [r22]
I170215 16:02:24.776041 11425 storage/replica_command.go:2397  [n1,s1,r18/1:/Table/5{1/1/9-2/1/2},@c42317d880] initiating a split of this range at key /Table/52/1/1 [r23]
I170215 16:02:24.783940 11721 storage/replica_command.go:2397  [n1,s1,r19/1:/Table/52/1/{5-8},@c421f18e00] initiating a split of this range at key /Table/52/1/7 [r24]
I170215 16:02:24.784787 11601 storage/replica_command.go:2397  [n1,s1,r21/1:/Table/52/1/{2-4},@c421e0d180] initiating a split of this range at key /Table/52/1/3 [r25]
I170215 16:02:24.786305 11646 storage/replica_command.go:2397  [n1,s1,r20/1:/{Table/52/1/8-Max},@c4217e2700] initiating a split of this range at key /Table/52/1/9 [r26]
I170215 16:02:24.796378 11692 storage/replica_command.go:2397  [n1,s1,r18/1:/Table/5{1/1/9-2/1/1},@c42317d880] initiating a split of this range at key /Table/52/1 [r27]
I170215 16:02:24.805112 11843 storage/replica_command.go:2397  [n1,s1,r19/1:/Table/52/1/{5-7},@c421f18e00] initiating a split of this range at key /Table/52/1/6 [r28]
I170215 16:02:24.864772 12020 storage/engine/rocksdb.go:336  opening rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader321891381"
I170215 16:02:24.879781 12013 storage/engine/rocksdb.go:336  opening rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader303221776"
I170215 16:02:24.879818 12016 storage/engine/rocksdb.go:336  opening rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader288279087"
I170215 16:02:24.879851 12014 storage/engine/rocksdb.go:336  opening rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader878049730"
I170215 16:02:24.880018 12018 storage/engine/rocksdb.go:336  opening rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader319576633"
I170215 16:02:24.881451 12017 storage/engine/rocksdb.go:336  opening rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader393438276"
I170215 16:02:24.885621 12019 storage/engine/rocksdb.go:336  opening rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader045438163"
I170215 16:02:24.885764 12015 storage/engine/rocksdb.go:336  opening rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader267090454"
I170215 16:02:24.886624 12011 storage/engine/rocksdb.go:336  opening rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader305639549"
I170215 16:02:24.891163 12012 storage/engine/rocksdb.go:336  opening rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader513024952"
I170215 16:02:25.014533 12020 storage/engine/rocksdb.go:401  closing rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader321891381"
I170215 16:02:25.088892 12019 storage/engine/rocksdb.go:401  closing rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader045438163"
I170215 16:02:25.089081 12016 storage/engine/rocksdb.go:401  closing rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader288279087"
I170215 16:02:25.095002 12018 storage/engine/rocksdb.go:401  closing rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader319576633"
I170215 16:02:25.098331 12011 storage/engine/rocksdb.go:401  closing rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader305639549"
I170215 16:02:25.098759 12012 storage/engine/rocksdb.go:401  closing rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader513024952"
I170215 16:02:25.099445 12017 storage/engine/rocksdb.go:401  closing rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader393438276"
I170215 16:02:25.101248 12013 storage/engine/rocksdb.go:401  closing rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader303221776"
I170215 16:02:25.104525 12015 storage/engine/rocksdb.go:401  closing rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader267090454"
I170215 16:02:25.104582 12014 storage/engine/rocksdb.go:401  closing rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader878049730"
I170215 16:02:28.634908 9135 storage/split_queue.go:98  [split,n1,s1,r18/1:/Table/5{1/1/9-2/1},@c42317d880] splitting at key /Table/52/0
I170215 16:02:28.634950 9135 storage/replica_command.go:2397  [split,n1,s1,r18/1:/Table/5{1/1/9-2/1},@c42317d880] initiating a split of this range at key /Table/52 [r29]
I170215 16:02:46.970615 9077 gossip/gossip.go:1192  [n1] node has connected to cluster via gossip
I170215 16:02:46.970737 9077 storage/stores.go:312  [n1] wrote 2 node addresses to persistent storage
I170215 16:02:58.383383 9485 gossip/gossip.go:1192  [n3] node has connected to cluster via gossip
I170215 16:02:58.383535 9485 storage/stores.go:312  [n3] wrote 2 node addresses to persistent storage
I170215 16:02:59.982753 9336 gossip/gossip.go:1192  [n2] node has connected to cluster via gossip
I170215 16:02:59.982941 9336 storage/stores.go:312  [n2] wrote 2 node addresses to persistent storage
W170215 16:03:10.229374 10092 storage/raft_transport.go:443  [n3] raft transport stream to node 2 failed: rpc error: code = 13 desc = transport is closing
W170215 16:03:10.229599 9692 storage/raft_transport.go:443  [n3] raft transport stream to node 1 failed: rpc error: code = 13 desc = transport is closing
W170215 16:03:10.229617 9336 gossip/gossip.go:1178  [n2] no incoming or outgoing connections
W170215 16:03:10.229626 9485 gossip/gossip.go:1178  [n3] no incoming or outgoing connections
I170215 16:03:10.229649 9233 vendor/google.golang.org/grpc/transport/http2_client.go:1123  transport: http2Client.notifyError got notified that the client transport was broken EOF.
I170215 16:03:10.229705 9307 vendor/google.golang.org/grpc/transport/http2_server.go:320  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:45838->127.0.0.1:58778: use of closed network connection
W170215 16:03:10.229737 9605 storage/raft_transport.go:443  [n2] raft transport stream to node 1 failed: EOF
I170215 16:03:10.229798 9321 vendor/google.golang.org/grpc/clientconn.go:766  grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:45838: operation was canceled"; Reconnecting to {127.0.0.1:45838 <nil>}
I170215 16:03:10.229821 9321 vendor/google.golang.org/grpc/clientconn.go:866  grpc: addrConn.transportMonitor exits due to: grpc: the connection is closing
W170215 16:03:10.230819 10168 storage/raft_transport.go:443  [n2] raft transport stream to node 3 failed: EOF

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/ccl/sqlccl/backup_test.go
+++ b/pkg/ccl/sqlccl/backup_test.go
@@ -12,7 +12,6 @@ import (
 	"bytes"
 	gosql "database/sql"
 	"fmt"
-	"hash/crc32"
 	"io/ioutil"
 	"math/rand"
 	"os"
@@ -69,7 +68,7 @@ func bankDataInsertStmts(count int) []string {
 				insert.WriteRune(',')
 			}
 			payload := randutil.RandBytes(rng, backupRestoreRowPayloadSize)
-			fmt.Fprintf(&insert, `(%d, %d, '%s')`, j, 0, payload)
+			fmt.Fprintf(&insert, `(%d, %d, 'initial-%s')`, j, 0, payload)
 		}
 		statements = append(statements, insert.String())
 	}
@@ -280,7 +279,9 @@ func TestBackupRestoreInterleaved(t *testing.T) {
 	})
 }
 
-func startBankTransfers(stopper *stop.Stopper, sqlDB *gosql.DB, numAccounts int) error {
+func startBackgroundWrites(
+	stopper *stop.Stopper, sqlDB *gosql.DB, wake chan<- struct{}, maxID int,
+) error {
 	rng, _ := randutil.NewPseudoRand()
 
 	for {
@@ -291,7 +292,7 @@ func startBankTransfers(stopper *stop.Stopper, sqlDB *gosql.DB, numAccounts int)
 			// Keep going.
 		}
 
-		account := rand.Intn(numAccounts)
+		id := rand.Intn(maxID)
 		payload := randutil.RandBytes(rng, backupRestoreRowPayloadSize)
 
 		updateFn := func() error {
@@ -301,12 +302,15 @@ func startBankTransfers(stopper *stop.Stopper, sqlDB *gosql.DB, numAccounts int)
 			default:
 				// Keep going.
 			}
-			_, err := sqlDB.Exec(`UPDATE bench.bank SET payload = $1 WHERE id = $2`,
-				payload, account)
+			_, err := sqlDB.Exec(`UPDATE bench.bank SET payload = $1 WHERE id = $2`, payload, id)
 			return err
 		}
 		if err := util.RetryForDuration(testutils.DefaultSucceedsSoonDuration, updateFn); err != nil {
 			return err
+		}
+		select {
+		case wake <- struct{}{}:
+		default:
 		}
 	}
 }
@@ -321,82 +325,59 @@ func TestBackupRestoreBank(t *testing.T) {
 	// TODO(dan): Actually invalidate the descriptor cache and delete this line.
 	defer sql.TestDisableTableLeases()()
 
-	const numAccounts = 10
-	const numTransferTasks = multiNode
+	const rows = 10
+	const numBackgroundTasks = multiNode
 	const backupRestoreIterations = 3
 
-	_, baseDir, tc, sqlDB, cleanupFn := backupRestoreTestSetup(t, multiNode, numAccounts)
+	_, baseDir, tc, sqlDB, cleanupFn := backupRestoreTestSetup(t, multiNode, rows)
 	defer cleanupFn()
 
-	for i := 0; i < numTransferTasks; i++ {
+	bgActivity := make(chan struct{})
+
+	for i := 0; i < numBackgroundTasks; i++ {
 		taskNum := i
 		tc.Stopper().RunWorker(func() {
 			// Use different sql gateways to make sure leasing is right.
-			err := startBankTransfers(tc.Stopper(), tc.Conns[taskNum%len(tc.Conns)], numAccounts)
+			err := startBackgroundWrites(tc.Stopper(), tc.Conns[taskNum%len(tc.Conns)], bgActivity, rows)
 			if err != nil {
 				t.Error(err)
 			}
 		})
 	}
 
-	waitForChecksumChange := func(currentChecksum uint32) uint32 {
-		var newChecksum uint32
-		testutils.SucceedsSoon(t, func() error {
-			crc := crc32.New(crc32.MakeTable(crc32.Castagnoli))
-			rows := sqlDB.Query(`SELECT payload FROM bench.bank`)
-			defer rows.Close()
-			var payload []byte
-			for rows.Next() {
-				if err := rows.Scan(&payload); err != nil {
-					t.Fatal(err)
-				}
-				if _, err := crc.Write(payload); err != nil {
-					t.Fatal(err)
-				}
-			}
-			if err := rows.Err(); err != nil {
-				t.Fatal(err)
-			}
-			newChecksum = crc.Sum32()
-			if newChecksum == currentChecksum {
-				return errors.Errorf("waiting for checksum to change %d", newChecksum)
-			}
-			return nil
-		})
-		return newChecksum
-	}
-
 	// Use the bench.bank table as a key (id), value (balance) table with a
-	// payload. Repeatedly run backups and restores while the transfer tasks are
+	// payload. Repeatedly run backups and restores while the background tasks are
 	// mutating the table concurrently. Wait in between each backup and each
 	// restore for the data to change.
-	var checksum uint32
 	for i := 0; i < backupRestoreIterations; i++ {
 		dir := filepath.Join(baseDir, strconv.Itoa(i))
 
+		<-bgActivity
+
 		// Set the id=balance invariant on each row and back up the table.
-		checksum = waitForChecksumChange(checksum)
-		_ = sqlDB.Exec(`UPDATE bench.bank SET balance = id`)
-		_ = sqlDB.Exec(fmt.Sprintf(`BACKUP DATABASE bench TO '%s'`, dir))
+		sqlDB.Exec(`UPDATE bench.bank SET balance = id`)
+
+		sqlDB.Exec(fmt.Sprintf(`BACKUP DATABASE bench TO '%s'`, dir))
+
+		<-bgActivity
 
 		// Break the id=balance invariant on every row in the current table.
 		// When we restore, it'll blow away this data, but if restore doesn't
 		// work we can tell by looking for these.
-		checksum = waitForChecksumChange(checksum)
-		_ = sqlDB.Exec(`UPDATE bench.bank SET balance = -1`)
+		sqlDB.Exec(`UPDATE bench.bank SET balance = -1`)
 
-		// Restore and wait for the new descriptor to be gossiped out. We can
-		// tell that this happened when id=balance holds again.
-		checksum = waitForChecksumChange(checksum)
-		_ = sqlDB.Exec(fmt.Sprintf(`RESTORE DATABASE bench FROM '%s'`, dir))
-		testutils.SucceedsSoon(t, func() error {
-			var failures int
-			sqlDB.QueryRow(`SELECT COUNT(id) FROM bench.bank WHERE id != balance`).Scan(&failures)
-			if failures > 0 {
-				return errors.Errorf("The bank is not in good order. Total failures: %d", failures)
-			}
-			return nil
-		})
+		<-bgActivity
+
+		// Restore and check id=balance holds again.
+		sqlDB.Exec(fmt.Sprintf(`RESTORE DATABASE bench FROM '%s'`, dir))
+
+		bad := sqlDB.QueryStr(`SELECT id, balance, payload FROM bench.bank WHERE id != balance`)
+		for _, r := range bad {
+			t.Errorf("bad row ID %s = bal %s (payload: %q)", r[0], r[1], r[2])
+		}
+		if len(bad) > 0 {
+			break
+		}
 	}
 }
 

--- a/pkg/ccl/sqlccl/backup_test.go
+++ b/pkg/ccl/sqlccl/backup_test.go
@@ -317,7 +317,6 @@ func startBackgroundWrites(
 
 func TestBackupRestoreBank(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	t.Skip("#13189")
 	if !storage.ProposerEvaluatedKVEnabled() {
 		t.Skip("command WriteBatch is not allowed without proposer evaluated KV")
 	}

--- a/pkg/ccl/storageccl/engineccl/mvcc.go
+++ b/pkg/ccl/storageccl/engineccl/mvcc.go
@@ -16,16 +16,16 @@ import (
 	"github.com/pkg/errors"
 )
 
-// MVCCIncrementalIterator iterates over the diff of the key range [start,end)
-// and time range [start,end). If a key was added or modified between startTime
-// and endTime, the iterator will position at the most recent version (before
-// endTime) of that key. If the key was most recently deleted, this is signalled
-// with an empty value.
+// MVCCIncrementalIterator iterates over the diff of the key range
+// [startKey,endKey) and time range [startTime,endTime). If a key was added or
+// modified between startTime and endTime, the iterator will position at the
+// most recent version (before endTime) of that key. If the key was most
+// recently deleted, this is signalled with an empty value.
 //
 // Expected usage:
 //    iter := NewMVCCIncrementalIterator(e)
 //    defer iter.Close()
-//    for iter.Reset(...); iter.Valid(); iter.Next() {
+//    for iter.Reset(startKey, endKey, ...); iter.Valid(); iter.Next() {
 //        [code using iter.Key() and iter.Value()]
 //    }
 //    if err := iter.Error(); err != nil {
@@ -119,7 +119,7 @@ func (i *MVCCIncrementalIterator) Next() {
 		}
 
 		if i.meta.Txn != nil {
-			if !i.meta.Timestamp.Less(i.endTime) {
+			if !i.endTime.Less(i.meta.Timestamp) {
 				i.err = &roachpb.WriteIntentError{
 					Intents: []roachpb.Intent{{Span: roachpb.Span{Key: metaKey.Key}, Status: roachpb.PENDING, Txn: *i.meta.Txn}},
 				}

--- a/pkg/ccl/storageccl/engineccl/mvcc_test.go
+++ b/pkg/ccl/storageccl/engineccl/mvcc_test.go
@@ -85,7 +85,7 @@ func TestMVCCIterateIncremental(t *testing.T) {
 		t.Run("", func(t *testing.T) {
 			iter := NewMVCCIncrementalIterator(e)
 			defer iter.Close()
-			for iter.Reset(keyMin, keyMax, ts0, ts3); iter.Valid(); iter.Next() {
+			for iter.Reset(keyMin, keyMax, startTime, endTime); iter.Valid(); iter.Next() {
 				// pass
 			}
 			if err := iter.Error(); !testutils.IsError(err, errString) {

--- a/pkg/ccl/storageccl/engineccl/mvcc_test.go
+++ b/pkg/ccl/storageccl/engineccl/mvcc_test.go
@@ -24,6 +24,49 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/uuid"
 )
 
+func iterateExpectErr(
+	e engine.Engine, startKey, endKey roachpb.Key, startTime, endTime hlc.Timestamp, errString string,
+) func(*testing.T) {
+	return func(t *testing.T) {
+		iter := NewMVCCIncrementalIterator(e)
+		defer iter.Close()
+		for iter.Reset(startKey, endKey, startTime, endTime); iter.Valid(); iter.Next() {
+			// pass
+		}
+		if err := iter.Error(); !testutils.IsError(err, errString) {
+			t.Fatalf("expected error %q but got %v", errString, err)
+		}
+	}
+}
+
+func assertEqualKVs(
+	e engine.Engine,
+	startKey, endKey roachpb.Key,
+	startTime, endTime hlc.Timestamp,
+	expected []engine.MVCCKeyValue,
+) func(*testing.T) {
+	return func(t *testing.T) {
+		iter := NewMVCCIncrementalIterator(e)
+		defer iter.Close()
+		var kvs []engine.MVCCKeyValue
+		for iter.Reset(startKey, endKey, startTime, endTime); iter.Valid(); iter.Next() {
+			kvs = append(kvs, engine.MVCCKeyValue{Key: iter.Key(), Value: iter.Value()})
+		}
+
+		if len(kvs) != len(expected) {
+			t.Fatalf("got %d kvs but expected %d: %v", len(kvs), len(expected), kvs)
+		}
+		for i := range kvs {
+			if !kvs[i].Key.Equal(expected[i].Key) {
+				t.Fatalf("%d key: got %v but expected %v", i, kvs[i].Key, expected[i].Key)
+			}
+			if !bytes.Equal(kvs[i].Value, expected[i].Value) {
+				t.Fatalf("%d value: got %x but expected %x", i, kvs[i].Value, expected[i].Value)
+			}
+		}
+	}
+}
+
 func TestMVCCIterateIncremental(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	ctx := context.Background()
@@ -53,57 +96,16 @@ func TestMVCCIterateIncremental(t *testing.T) {
 		return engine.MVCCKeyValue{Key: engine.MVCCKey{Key: key, Timestamp: ts}, Value: value}
 	}
 
-	assertEqualKVs := func(
-		startKey, endKey roachpb.Key, startTime, endTime hlc.Timestamp,
-		expected []engine.MVCCKeyValue,
-	) {
-		t.Run("", func(t *testing.T) {
-			iter := NewMVCCIncrementalIterator(e)
-			defer iter.Close()
-			var kvs []engine.MVCCKeyValue
-			for iter.Reset(startKey, endKey, startTime, endTime); iter.Valid(); iter.Next() {
-				kvs = append(kvs, engine.MVCCKeyValue{Key: iter.Key(), Value: iter.Value()})
-			}
-
-			if len(kvs) != len(expected) {
-				t.Fatalf("got %d kvs but expected %d: %v", len(kvs), len(expected), kvs)
-			}
-			for i := range kvs {
-				if !kvs[i].Key.Equal(expected[i].Key) {
-					t.Fatalf("%d key: got %v but expected %v", i, kvs[i].Key, expected[i].Key)
-				}
-				if !bytes.Equal(kvs[i].Value, expected[i].Value) {
-					t.Fatalf("%d value: got %x but expected %x", i, kvs[i].Value, expected[i].Value)
-				}
-			}
-		})
-	}
-
-	iterateExpectErr := func(
-		startKey, endKey roachpb.Key, startTime, endTime hlc.Timestamp, errString string,
-	) {
-		t.Run("", func(t *testing.T) {
-			iter := NewMVCCIncrementalIterator(e)
-			defer iter.Close()
-			for iter.Reset(keyMin, keyMax, startTime, endTime); iter.Valid(); iter.Next() {
-				// pass
-			}
-			if err := iter.Error(); !testutils.IsError(err, errString) {
-				t.Fatalf("expected error %q but got %v", errString, err)
-			}
-		})
-	}
-
 	kv1_1_1 := makeKVT(testKey1, testValue1, ts1)
 	kv1_4_4 := makeKVT(testKey1, testValue4, ts4)
-	kv2_2_1 := makeKVT(testKey1, testValue2, ts2)
+	kv1_2_2 := makeKVT(testKey1, testValue2, ts2)
 	kv2_2_2 := makeKVT(testKey2, testValue3, ts2)
 	kv1_3Deleted := makeKVT(testKey1, nil, ts3)
 	kvs := func(kvs ...engine.MVCCKeyValue) []engine.MVCCKeyValue { return kvs }
 
-	assertEqualKVs(keyMin, keyMax, ts0, ts3, nil)
+	t.Run("empty", assertEqualKVs(e, keyMin, keyMax, ts0, ts3, nil))
 
-	for _, kv := range kvs(kv1_1_1, kv2_2_1, kv2_2_2) {
+	for _, kv := range kvs(kv1_1_1, kv1_2_2, kv2_2_2) {
 		v := roachpb.Value{RawBytes: kv.Value}
 		if err := engine.MVCCPut(ctx, e, nil, kv.Key.Key, kv.Key.Timestamp, v, nil); err != nil {
 			t.Fatal(err)
@@ -111,22 +113,22 @@ func TestMVCCIterateIncremental(t *testing.T) {
 	}
 
 	// Exercise time ranges.
-	assertEqualKVs(keyMin, keyMax, ts1, ts1, nil)
-	assertEqualKVs(keyMin, keyMax, ts1, ts2, kvs(kv1_1_1))
-	assertEqualKVs(keyMin, keyMax, ts1, tsMax, kvs(kv2_2_1, kv2_2_2))
-	assertEqualKVs(keyMin, keyMax, ts2, ts2, nil)
-	assertEqualKVs(keyMin, keyMax, ts2, ts3, kvs(kv2_2_1, kv2_2_2))
-	assertEqualKVs(keyMin, keyMax, ts3, ts3, nil)
+	t.Run("ts 1-1", assertEqualKVs(e, keyMin, keyMax, ts1, ts1, nil))
+	t.Run("ts 1-2", assertEqualKVs(e, keyMin, keyMax, ts1, ts2, kvs(kv1_1_1)))
+	t.Run("ts 1-∞", assertEqualKVs(e, keyMin, keyMax, ts1, tsMax, kvs(kv1_2_2, kv2_2_2)))
+	t.Run("ts 2-2", assertEqualKVs(e, keyMin, keyMax, ts2, ts2, nil))
+	t.Run("ts 2-3", assertEqualKVs(e, keyMin, keyMax, ts2, ts3, kvs(kv1_2_2, kv2_2_2)))
+	t.Run("ts 3-3", assertEqualKVs(e, keyMin, keyMax, ts3, ts3, nil))
 
 	// Exercise key ranges.
-	assertEqualKVs(testKey1, testKey1, ts0, tsMax, nil)
-	assertEqualKVs(testKey1, testKey2, ts0, tsMax, kvs(kv2_2_1))
+	t.Run("kv 1-1", assertEqualKVs(e, testKey1, testKey1, ts0, tsMax, nil))
+	t.Run("kv 1-2", assertEqualKVs(e, testKey1, testKey2, ts0, tsMax, kvs(kv1_2_2)))
 
 	// Exercise deletion.
 	if err := engine.MVCCDelete(ctx, e, nil, testKey1, ts3, nil); err != nil {
 		t.Fatal(err)
 	}
-	assertEqualKVs(keyMin, keyMax, ts0, tsMax, kvs(kv1_3Deleted, kv2_2_2))
+	t.Run("del", assertEqualKVs(e, keyMin, keyMax, ts0, tsMax, kvs(kv1_3Deleted, kv2_2_2)))
 
 	// Exercise intent handling.
 	txn1ID := uuid.MakeV4()
@@ -151,9 +153,11 @@ func TestMVCCIterateIncremental(t *testing.T) {
 	if err := engine.MVCCPut(ctx, e, nil, txn2.TxnMeta.Key, txn2.TxnMeta.Timestamp, txn2Val, &txn2); err != nil {
 		t.Fatal(err)
 	}
-	iterateExpectErr(testKey1, testKey1, ts0, tsMax, "conflicting intents")
-	iterateExpectErr(testKey2, testKey2, ts0, tsMax, "conflicting intents")
-	assertEqualKVs(keyMin, keyMax, ts0, ts4, nil)
+	t.Run("intents1",
+		iterateExpectErr(e, testKey1, testKey1.PrefixEnd(), ts0, tsMax, "conflicting intents"))
+	t.Run("intents2",
+		iterateExpectErr(e, testKey2, testKey2.PrefixEnd(), ts0, tsMax, "conflicting intents"))
+	t.Run("intents3", assertEqualKVs(e, keyMin, keyMax, ts0, ts4, nil))
 
 	intent1 := roachpb.Intent{Span: roachpb.Span{Key: testKey1}, Txn: txn1.TxnMeta, Status: roachpb.COMMITTED}
 	if err := engine.MVCCResolveWriteIntent(ctx, e, nil, intent1); err != nil {
@@ -163,5 +167,5 @@ func TestMVCCIterateIncremental(t *testing.T) {
 	if err := engine.MVCCResolveWriteIntent(ctx, e, nil, intent2); err != nil {
 		t.Fatal(err)
 	}
-	assertEqualKVs(keyMin, keyMax, ts0, tsMax, kvs(kv1_4_4, kv2_2_2))
+	t.Run("intents4", assertEqualKVs(e, keyMin, keyMax, ts0, tsMax, kvs(kv1_4_4, kv2_2_2)))
 }

